webapp/src/snippetBuilder.tsx

```diff
@@ -0,0 +1,339 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+import * as React from "react";
+import * as data from "./data";
+import * as sui from "./sui";
+import * as md from "./marked";
+import * as compiler from './compiler';
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+export interface CreateSnippetBuilderState {
+    visible?: boolean;
+    output?: string;
+    projectView?: pxt.editor.IProjectView;
+    answers?: any; // Will be typed once more clearly defined
+    currentQuestion: number;
+    defaults: any; // Will be typed once more clearly defined
+    mainWorkspace?: Blockly.Workspace;
+    config?: any; // Will be a config type
+}
+
+export class CreateSnippetBuilder extends data.Component<ISettingsProps, CreateSnippetBuilderState> {
```

Comment:
Should this be called just "SnippetBuilder" ?

Also, would you mind adding a descriptive comment explaining the purpose of this react component? That'll help future developers. You'll notice that our code base doesn't have a lot of descriptive comments like this but we're trying to slowly get the code base into a more readable state so that also means new code should have some top level comments.

Reply:
Yeah I changed the naming to make more sense.